src/main/java/com/ge/predix/uaa/token/lib/ZacTokenService.java

```diff
@@ -72,5 +93,10 @@ public void flushIssuerCache(final String zoneId) {
     public void afterPropertiesSet() throws Exception {
         long timeToLiveMillis = this.issuersTtlSeconds * 1000;
         this.tokenServicesMap = new PassiveExpiringMap<>(timeToLiveMillis);
+
```

Comment:
this should suffice 
`if (  StringUtils.isBlank(this.serviceBaseDomain) || StringUtils.isBlank(this.serviceZoneHeaders) {
.....
} `

consider IllegalArgumentException, because in this case there is no request yet.